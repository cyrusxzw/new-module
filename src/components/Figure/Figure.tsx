import React, { FC } from 'react';
import cx from 'classnames';
import type { FigureProps } from './Figure.types';
import styles from './Figure.module.css';

const Figure: FC<FigureProps> = ({
  caption,
  children,
  className,
  hasCaptionBorder,
  hasCaptionOffset,
  id,
  theme = 'dark',
}) => {
  const classSet = cx(styles.base, styles[theme], className);

  const captionWrapperClassSet = cx(styles.captionWrapper, {
    [styles.border]: hasCaptionBorder && caption,
    [styles.captionOffset]: hasCaptionOffset && caption,
  });

  return (
    <figure className={classSet} id={id}>
      {children}
      {caption && (
        <figcaption className={captionWrapperClassSet}>
          <div className={styles.caption}>{caption}</div>
        </figcaption>
      )}
    </figure>
  );
};

export { Figure };
