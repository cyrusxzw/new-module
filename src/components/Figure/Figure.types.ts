type FigureProps = {
  caption?: string;
  children?: React.ReactNode;
  className?: string;
  hasCaptionBorder?: boolean;
  hasCaptionOffset?: boolean;
  id?: string;
  theme?: string;
  foo?: boolean;
};

export type { FigureProps };
