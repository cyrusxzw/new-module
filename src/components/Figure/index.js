import { Figure } from './Figure.tsx';

export { Figure };
