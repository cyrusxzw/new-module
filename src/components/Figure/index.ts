import { Figure } from './Figure';

export { Figure };
