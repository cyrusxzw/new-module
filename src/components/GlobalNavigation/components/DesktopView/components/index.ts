export { CloseButton } from './CloseButton';
export { CollectionImage } from './CollectionImage';
export { CollectionLayout } from './CollectionLayout';
export { CollectionList } from './CollectionList';
export { MenuItem } from './MenuItem';
export { Panel } from './Panel';
export { PrimaryMenu } from './PrimaryMenu';
export { SecondaryMenu } from './SecondaryMenu';
