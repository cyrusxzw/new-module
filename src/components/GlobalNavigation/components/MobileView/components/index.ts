export { ArticleCard } from './ArticleCard';
export { ArticleList } from './ArticleList';
export { Collection } from './Collection';
export { Header } from './Header';
export { Link } from './Link';
export { ListItem } from './ListItem';
export { NestedCollection } from './NestedCollection';
export { PrimaryNavigation } from './PrimaryNavigation';
export { PromotionCard } from './PromotionCard';
export { SecondaryNavigation } from './SecondaryNavigation';
