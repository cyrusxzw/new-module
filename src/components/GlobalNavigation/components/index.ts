export { DesktopView } from './DesktopView';
export { MobileView } from './MobileView';
export { TabletView } from './TabletView';
