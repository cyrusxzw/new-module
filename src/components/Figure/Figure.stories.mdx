import { Meta, Story, Canvas, ArgsTable } from '@storybook/addon-docs/blocks';
import * as knobs from '@storybook/addon-knobs';
import { Figure } from './Figure';
import { Image } from '~/components/Image';

<Meta title="Components/Figure" component={Figure} />

# Figure

<Canvas>
  <Story name="Base component">
    <Figure
      caption={knobs.text('caption', 'Lorem ipsum dolor sit amet', 'Copy')}
      hasCaptionBorder={knobs.boolean('hasCaptionBorder', true, 'Presentation')}
      hasCaptionOffset={knobs.boolean('hasCaptionOffset', true, 'Presentation')}
      theme={knobs.select('theme', ['dark', 'light'], 'dark', 'Presentation')}
    >
      <Image
        altText="Lorem ipsum dolor sit amet"
        sizes={{
          large: knobs.text(
            'large',
            'https://www.aesop.com/medias/GelCleanser-640x640.png?context=bWFzdGVyfGltYWdlc3w1MzE5NTl8aW1hZ2UvcG5nfGltYWdlcy9oOTUvaGNkLzg4MDc3OTc4NTAxNDIucG5nfGM3NTNhYjdkMWNiOTRhNWZhNDA0NWExZWJiODMxN2VhNzEyY2ZkNGUyYzc4ZWVkZDRjYTIzMzZmZTRlOGYxNjM',
            'Image',
          ),
          medium: knobs.text(
            'medium',
            'https://www.aesop.com/medias/GelCleanser-640x640.png?context=bWFzdGVyfGltYWdlc3w1MzE5NTl8aW1hZ2UvcG5nfGltYWdlcy9oOTUvaGNkLzg4MDc3OTc4NTAxNDIucG5nfGM3NTNhYjdkMWNiOTRhNWZhNDA0NWExZWJiODMxN2VhNzEyY2ZkNGUyYzc4ZWVkZDRjYTIzMzZmZTRlOGYxNjM',
            'Image',
          ),
          small: knobs.text(
            'small',
            'https://www.aesop.com/medias/GelCleanser-640x640.png?context=bWFzdGVyfGltYWdlc3w1MzE5NTl8aW1hZ2UvcG5nfGltYWdlcy9oOTUvaGNkLzg4MDc3OTc4NTAxNDIucG5nfGM3NTNhYjdkMWNiOTRhNWZhNDA0NWExZWJiODMxN2VhNzEyY2ZkNGUyYzc4ZWVkZDRjYTIzMzZmZTRlOGYxNjM',
            'Image',
          ),
        }}
      />
    </Figure>
  </Story>
</Canvas>

## API

<ArgsTable of={Figure} />
