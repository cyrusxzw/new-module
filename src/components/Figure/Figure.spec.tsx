import * as React from 'react';
import { render } from '@testing-library/react';
import { Figure } from './Figure';
import { Image } from '~/components/Image';

describe('<Figure />', () => {
  it('should be defined', () => {
    expect(Figure).toBeDefined();
  });

  it('should render base component correctly', () => {
    const { container } = render(
      <Figure caption="About Aesop" id="about">
        <Image
          altText="Image Alt Text"
          large="https://release.aesop-web-ui.aesopdigital.io/images/products/Skin_Perfect_Facial_Hydrating_Cream_60mL_large.png"
        />
      </Figure>,
    );

    expect(container).toMatchSnapshot();
  });
});
