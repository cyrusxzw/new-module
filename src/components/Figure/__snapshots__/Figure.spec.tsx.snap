// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Figure /> should render base component correctly 1`] = `
<div>
  <figure
    class="base dark"
    id="about"
  >
    <picture
      class="base picture dark"
    >
      <source
        media="(min-width: 1025px)"
        srcset="https://release.aesop-web-ui.aesopdigital.io/images/products/Skin_Perfect_Facial_Hydrating_Cream_60mL_large.png"
      />
      <img
        alt="Image Alt Text"
        loading="lazy"
      />
    </picture>
    <figcaption
      class="captionWrapper"
    >
      <div
        class="caption"
      >
        About Aesop
      </div>
    </figcaption>
  </figure>
</div>
`;
